pkg/distroidparser/idparser.go

```diff
@@ -0,0 +1,57 @@
+package distroidparser
+
+import (
+	"github.com/osbuild/images/pkg/distro"
+	"github.com/osbuild/images/pkg/distro/fedora"
+	"github.com/osbuild/images/pkg/distro/rhel7"
+	"github.com/osbuild/images/pkg/distro/rhel8"
+	"github.com/osbuild/images/pkg/distro/rhel9"
+)
+
+type ParserFunc func(idStr string) (*distro.ID, error)
+
+// Parser is a list of distro-specific idStr parsers.
+type Parser struct {
+	parsers []ParserFunc
+}
+
+func NewParser(parsers ...ParserFunc) *Parser {
```

Comment:
Would it make sense to rename this to plain `New()`?
```
parser := distroidparser.New(...)
```

Reply:
Done